cmd/influx/measurement_schema.go

```diff
@@ -0,0 +1,247 @@
+					Destination: &params.ColumnsFile,
+				},
+				&cli.GenericFlag{
+					Name:        "columns-format",
+					Usage:       "The format of the columns file. \"auto\" will attempt to guess the format.",
+					DefaultText: "auto",
+					Value:       &params.ColumnsFormat,
+				},
+				&cli.BoolFlag{
+					Name:        "extended-output",
+					Usage:       "Print column information for each measurement",
+					Aliases:     []string{"x"},
+					Destination: &params.ExtendedOutput,
+				},
+			)...,
+		),
+		Action: func(ctx *cli.Context) error {
+			var (
+				r    io.Reader
+				name string
+			)
+
+			arg := ctx.Args().First()
+			if arg == "-" || params.ColumnsFile == "" {
+				// stdio
+				r = ctx.App.Reader
+				name = "stdio"
+			} else {
+				data, err := os.ReadFile(params.ColumnsFile)
+				if err != nil {
+					return fmt.Errorf("unable to read file %q: %w", params.ColumnsFile, err)
+				}
+				r = bytes.NewReader(data)
+				name = params.ColumnsFile
+			}
+
+			reader, err := params.ColumnsFormat.FormatFn(name)
+			if err != nil {
+				return err
+			}
+
+			cols, err := reader(r)
+			if err != nil {
+				return fmt.Errorf("unable to decode file %q: %w", name, err)
+			}
+
+			return getMSClient(ctx).Create(ctx.Context, measurement_schema.CreateParams{
+				OrgBucketParams: params.OrgBucketParams,
+				Name:            params.Name,
+				Columns:         cols,
+				ExtendedOutput:  params.ExtendedOutput,
+			})
+		},
+	}
+}
+
+func newMeasurementSchemaUpdateCmd() *cli.Command {
+	var params struct {
+		internal.OrgBucketParams
+		ID             influxid.ID
+		Name           string
+		ColumnsFile    string
+		ColumnsFormat  schema.Format
+		ExtendedOutput bool
+	}
+	return &cli.Command{
+		Name:   "update",
+		Usage:  "Update a measurement schema for a bucket",
+		Before: withMSClient(),
+		Flags: append(
+			commonFlags,
+			append(
+				getOrgBucketFlags(&params.OrgBucketParams),
+				&cli.GenericFlag{
+					Name:  "id",
+					Usage: "ID of the measurement",
+					Value: &params.ID,
+				},
+				&cli.StringFlag{
+					Name:        "name",
+					Usage:       "Name of the measurement",
+					Destination: &params.Name,
+				},
+				&cli.StringFlag{
+					Name:        "columns-file",
+					Usage:       "A path referring to list of column definitions",
+					Destination: &params.ColumnsFile,
+				},
+				&cli.GenericFlag{
+					Name:        "columns-format",
+					Usage:       "The format of the columns file. \"auto\" will attempt to guess the format.",
+					DefaultText: "auto",
+					Value:       &params.ColumnsFormat,
+				},
+				&cli.BoolFlag{
+					Name:        "extended-output",
+					Usage:       "Print column information for each measurement",
+					Aliases:     []string{"x"},
+					Destination: &params.ExtendedOutput,
+				},
+			)...,
+		),
+		Action: func(ctx *cli.Context) error {
+			var (
+				r    io.Reader
+				name string
+			)
+
+			arg := ctx.Args().First()
+			if arg == "-" || params.ColumnsFile == "" {
```

Comment:
I've moved it to a shared function in the `measurement_schema` package.

Reply:
That's a fair point. My thinking wasn't so much to make `internal/` entirely abstracted from being a CLI (for example, everything is driven through the `CLI` struct), but instead to abstract it away from `urfave/cli` in case we ended up needing to switch back to cobra. Not sure if that's a useful enough separation to maintain, but it does allow for things like stdin detection to be unit-tested more easily.